alpha/scripts/analyzeEmailFieldForPartner.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+require_once(__DIR__ . '/bootstrap.php');
+
+// parse the command line
+if($argc<3)
+{
+	die("Usage: php " . $argv[0] . " <partner id> <user type: admin | user> <realRun | dryRun>\n");
+}
+
+$partnerId = $argv[1];
+$userType = $argv[2];
+$dryRun = $argv[3] ? $argv[3] : 'dryRun';
+var_dump($partnerId, $userType, $dryRun);
+
+if (!PartnerPeer::retrieveByPK($partnerId))
+{
+	die("Please enter a valid partner Id!\n");
+}
+
+if (is_null($userType) || ($userType != 'admin' && $userType != 'user'))
+{
+	die("Please specify if looking for admin users (admin) or regular users (user)\n");
+}
+
+$isAdmin = 0;
+if ($userType === 'admin')
+{
+	$isAdmin = 1;
+}
+
+
+
+function countUsers($partnerId, $isAdmin, $emailFieldValue)
+{
+	$emailCriteria = new Criteria();
+	$emailCriteria->add(kuserPeer::PARTNER_ID, $partnerId, Criteria::EQUAL);
+	$emailCriteria->add(kuserPeer::STATUS, KuserStatus::ACTIVE);
+	$emailCriteria->add(kuserPeer::IS_ADMIN, $isAdmin);
+	if($emailFieldValue == Criteria::ISNULL) // counting users without email value
+	{
+		$emailCriteria->add(kuserPeer::EMAIL, null, Criteria::ISNULL);
+	}
+	else
+	{
+		$emailCriteria->add(kuserPeer::EMAIL, null, Criteria::ISNOTNULL);
+	}
+
+	return kuserPeer::doCount($emailCriteria);
+}
+
+function noEmailPercentage($noEmailUsersCount, $allUsersCount)
+{
+	return (int)(($noEmailUsersCount * 100)/$allUsersCount);
+}
+
+function getUsersWithEmail($partnerId, $isAdmin)
```

Comment:
why we need one query for count and one for getting the objects? cant we get the objects from the start and just see the size of the array?